compat/win32_maperror.cc

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/win32_maperror.h"
+
+#if _SQUID_WINDOWS_ || _SQUID_MINGW_ && !_SQUID_CYGWIN_
+
+#if HAVE_WINDOWS_H
+#include <windows.h>
+#endif
+// for _doserrno
+#if HAVE_STDLIB_H
+#include <stdlib.h>
+#endif
```

Comment:
We are in C++ territory here:

```suggestion
#include <cstdlib>
```